d3a_interface/device_validator.py

```diff
@@ -0,0 +1,245 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import json
+from d3a_interface.constants_limits import ConstSettings
+
+
+GeneralSettings = ConstSettings.GeneralSettings
+LoadSettings = ConstSettings.LoadSettings
+PvSettings = ConstSettings.PVSettings
+StorageSettings = ConstSettings.StorageSettings
+
+
+def validate_load_device(**kwargs):
+    if ("avg_power_W" in kwargs and kwargs["avg_power_W"] is not None) and \
+            not (LoadSettings.AVG_POWER_RANGE.min <= kwargs["avg_power_W"] <=
+                 LoadSettings.AVG_POWER_RANGE.max):
+        return json.dumps({"mis_configuration": [f"avg_power_W should be in between "
```

Comment:
Why not raise an exception here ? That way it will be far easier for the calling code to just catch the exception. From the d3a and d3a-web PRs, I saw that a boolean variable was returned and an exception was raised according to the variable. If you raise a ValueError (or, even better, a new exception type eg. D3AValidateException) you will not have to care about raising an exception on the calling code at least on d3a. On d3a-web you will just have to catch this exception and re-raise it as a GraphQLError.